src/app/components/category-products/category-products.component.ts

```diff
@@ -7,5 +18,122 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   encapsulation: ViewEncapsulation.Emulated,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class CategoryProductsComponent {
+export class CategoryProductsComponent implements AfterViewInit {
+  readonly categoryList$: Observable<CategoryModel[]> = this._categoriesService.getAllCategories();
+  readonly categoryId$: Observable<string> = this._activatedRoute.params.pipe(
+    map((params) => params['categoryId']),
+    shareReplay(1)
+  );
+  readonly queryParams$: Observable<QueryParamsQueryModel> = this._activatedRoute.queryParams.pipe(
+    map((params) => ({
+      sort: params['sort'] ?? 'featureValue',
+      order: params['order'] ?? 'desc'
+    })),
+    shareReplay(1),
+    tap((data) => this.setControlFromQueryParams(data))
+  );
+
+  readonly sortingOpts$: Observable<SortOptionModel[]> = of([
+    { display: 'Featured', key: 'featureValue;desc', order: 'desc' },
+    { display: 'Price Low to High', key: 'price;asc', order: 'asc' },
```

Comment:
if you have key, then you do not need order, or keep only the property in key

Reply:
property order deleted